Update lex_utf8.txt to modify Corsican translation

Patriccollu opened the pull request:
Hi GeneWeb team,

This is an update of **Corsican** translation to take the new strings into account. In addition, several apostrophe signs and some quotation marks have been modified.

Best regards,
Patriccollu di Santa Maria è Sichè.

Patriccollu committed:
Update lex_utf8.txt to modify Corsican translation

Adding the new strings and modifying several apostrophe signs and some quotation marks.

Patriccollu commented:
Hello,

During my tests I ran into some problems when specifying **square brackets** and **vertical bar** within some translations strings. It looks to not work as explained on the [wiki](https://geneweb.tuxfamily.org/wiki/lexicon/fr)

Please can you tell me where to get the latest (or at least a recent) **gw7 executable** for Windows? I would like to do further tests before opening an issue.

Thanks,
Patriccollu.

hgouraud commented:
the square brackets and vertical bar trick applies only in the a_of_b translation!
I am looking at what it would take to have it applied everywhere, but this is tricky!!

hgouraud commented:
I have looked in detail this de/d' issue.
The situation is rather confused, but can be summarized as follows:
- in general, this situation is NOT treated, except in two cases
a/ in the %1 of %2 case
b/ in the (date) case

The situation is confused by the fact that those two cases do not use the same syntax!! (We will change that to a uniform syntax):
%1 of %2 uses the d[e |’] syntax
(date) uses the d[e’] syntax

I hope this explains what you have observed.
I note that you had (at least) one instance where you wanted to use this possibility outside of the two treated cases. Is this important?
I ma looking at what it would take to handle de/d’ in all cases, but this seems rather complex

Henri

> Le 11 août 2019 à 18:13, Patriccollu <notifications@github.com> a écrit :
> 
> Hi GeneWeb team,
> 
> This is an update of Corsican translation to take the new strings into account. In addition, several apostrophe signs and some quotation marks have been modified.
> 
> Best regards,
> Patriccollu di Santa Maria è Sichè.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/geneweb/geneweb/pull/856 <https://github.com/geneweb/geneweb/pull/856>
> Commit Summary
> 
> Update lex_utf8.txt to modify Corsican translation
> File Changes
> 
> M hd/lang/lex_utf8.txt <https://github.com/geneweb/geneweb/pull/856/files#diff-0> (222)
> Patch Links:
> 
> https://github.com/geneweb/geneweb/pull/856.patch <https://github.com/geneweb/geneweb/pull/856.patch>
> https://github.com/geneweb/geneweb/pull/856.diff <https://github.com/geneweb/geneweb/pull/856.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/geneweb/geneweb/pull/856?email_source=notifications&email_token=ABNMQTJ7XRBGBGF4PN3PF6TQEA3DJA5CNFSM4IK4EZLKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HES27YA>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABNMQTIRR77TA6GJ75RP2MDQEA3DJANCNFSM4IK4EZLA>.
>

Patriccollu commented:
@hgouraud 
Hi Henri,

Thanks a lot for your research and findings which explain what I've noticed, and will help me modifying the strings accordingly!

Actually my first idea was to use this interesting syntax for a maximum of strings, such as:

```
    relations of
    residence%t to
    witness to events of
    witness to events of
    %t witnessed the %s after his/her death
    %t witnessed the %s before his/her birth
    %t's sex is not coherent with his/her relations
    age at %s birth date
    direct ancestor of %s
    insufficient revision history for %s
    marriage had occurred after the death of %t
    marriage had occurred before the birth of %t
    marriage of %t after his/her death
    marriage of %t before his/her birth
    on %s's side
    relations of
    the following children of %t and %t are born very close
    the following children of %t and %t are not in order
    witness at marriage of %s and %s
    witness to events of
```

Nevertheless, I checked the lex_utf8.txt and noticed that this syntax wasn't used a lot! So I'm not sure whether it worth doing a lot of work to allow using this syntax on other strings if it is not much used!

By the way, do you know where I can download a more recent gw7 executable for Windows?

Thanks,
Patriccollu.

Patriccollu commented:
@hgouraud 
Hi Henri,

I made some additional tests using my old gw7 version.

For %1 of %2 string, I coded: `co: %1 d[i |']%2`

With command: `http://localhost:2317/basename?lang=co;templ=templm;m=A;p=anatole;n=casata`
The title displayed is correct: `Antenati d'Anatole CASATA`

With command: `http://localhost:2317/basename?lang=co&templ=templm&m=A&p=anatole;n=casata&maxv=5&v=4&bd=0&color=&t=G&siblings=on&alias=on&spouse=on&parents=on&rel=on&notes=on&src=on`

The title displayed is incorrect: `Antenati di | Anatole CASATA nnnn-nnnn`

Note that if I use another character (indicated in the wiki) for the quote such as: `co: %1 d[i |’]%2`

Then the same above commands give these results, both incorrect:
```
	Antenati d�Anatole CASATA
	Antenati di |� Anatole CASATA nnnn-nnnn
```

For the (date) string, it works fine as long as I use `'`

If I use `’` all the dates are incorrect:
```
	natu u 24 d[i’]ghjennaghju
	maritatu u 6 d[i’]ghjugnu
	natu u 9 d[i’]uttobre
```

In summary, the best display is obtained with:
```
	co: %1 d[i |']%2
	co: 1<sup>u</sup> d[i']%m %y/%d d[i']%m %y/d[i']%m %y/in u %y
```

Thanks,
Patriccollu.

hgouraud commented:
the simple quote ‘ is an ascii character. The apostrophe ’ is a two byte utf8 character.
This explains the problem you encounter when using ’!!
This needs to be corrected.

As for the title misbehaving, I will look further! Quite strange indeed.

Henri

> Le 15 août 2019 à 15:29, Patriccollu <notifications@github.com> a écrit :
> 
> @hgouraud <https://github.com/hgouraud>
> Hi Henri,
> 
> I made some additional tests using my old gw7 version.
> 
> For %1 of %2 string, I coded: co: %1 d[i |']%2
> 
> With command: http://localhost:2317/basename?lang=co;templ=templm;m=A;p=anatole;n=casata
> The title displayed is correct: Antenati d'Anatole CASATA
> 
> With command: http://localhost:2317/basename?lang=co&templ=templm&m=A&p=anatole;n=casata&maxv=5&v=4&bd=0&color=&t=G&siblings=on&alias=on&spouse=on&parents=on&rel=on&notes=on&src=on
> 
> The title displayed is incorrect: Antenati di | Anatole CASATA nnnn-nnnn
> 
> Note that if I use another character (indicated in the wiki) for the quote such as: co: %1 d[i |’]%2
> 
> Then the same above commands give these results, both incorrect:
> 
> 	Antenati d�Anatole CASATA
> 	Antenati di |� Anatole CASATA nnnn-nnnn
> For the (date) string, it works fine as long as I use '
> 
> If I use ’ all the dates are incorrect:
> 
> 	natu u 24 d[i’]ghjennaghju
> 	maritatu u 6 d[i’]ghjugnu
> 	natu u 9 d[i’]uttobre
> In summary, the best display is obtained with:
> 
> 	co: %1 d[i |']%2
> 	co: 1<sup>u</sup> d[i']%m %y/%d d[i']%m %y/d[i']%m %y/in u %y
> Thanks,
> Patriccollu.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/geneweb/geneweb/pull/856?email_source=notifications&email_token=ABNMQTLM5XSBFYILUHXV4SLQEVKZ5A5CNFSM4IK4EZLKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4LZWPQ#issuecomment-521640766>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABNMQTIXP4JSNG6W6QTH5X3QEVKZ5ANCNFSM4IK4EZLA>.
>

hgouraud commented:
I cannot replicate your behavior!
co: %1 d[i |’]%2
Antenati d’Aaaa Xxxx Dddd Gouraud

Which commit are you running?
Henri

> Le 15 août 2019 à 15:29, Patriccollu <notifications@github.com> a écrit :
> 
> @hgouraud <https://github.com/hgouraud>
> Hi Henri,
> 
> I made some additional tests using my old gw7 version.
> 
> For %1 of %2 string, I coded: co: %1 d[i |']%2
> 
> With command: http://localhost:2317/basename?lang=co;templ=templm;m=A;p=anatole;n=casata
> The title displayed is correct: Antenati d'Anatole CASATA
> 
> With command: http://localhost:2317/basename?lang=co&templ=templm&m=A&p=anatole;n=casata&maxv=5&v=4&bd=0&color=&t=G&siblings=on&alias=on&spouse=on&parents=on&rel=on&notes=on&src=on
> 
> The title displayed is incorrect: Antenati di | Anatole CASATA nnnn-nnnn
> 
> Note that if I use another character (indicated in the wiki) for the quote such as: co: %1 d[i |’]%2
> 
> Then the same above commands give these results, both incorrect:
> 
> 	Antenati d�Anatole CASATA
> 	Antenati di |� Anatole CASATA nnnn-nnnn
> For the (date) string, it works fine as long as I use '
> 
> If I use ’ all the dates are incorrect:
> 
> 	natu u 24 d[i’]ghjennaghju
> 	maritatu u 6 d[i’]ghjugnu
> 	natu u 9 d[i’]uttobre
> In summary, the best display is obtained with:
> 
> 	co: %1 d[i |']%2
> 	co: 1<sup>u</sup> d[i']%m %y/%d d[i']%m %y/d[i']%m %y/in u %y
> Thanks,
> Patriccollu.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/geneweb/geneweb/pull/856?email_source=notifications&email_token=ABNMQTLM5XSBFYILUHXV4SLQEVKZ5A5CNFSM4IK4EZLKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4LZWPQ#issuecomment-521640766>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABNMQTIXP4JSNG6W6QTH5X3QEVKZ5ANCNFSM4IK4EZLA>.
>

hgouraud commented:
By the way, the d[e |’] syntax caters for utf_8 characters when the d[e’] syntax does not. The latter considers that there are only two ascii characters between the brackets []

Patriccollu commented:
I'm using the version 7.00 alpha for [Windows (x86)](http://download.tuxfamily.org/geneweb/release/7.00a/gw-7.00-alpha-win.exe) downloaded from the [wiki](https://geneweb.tuxfamily.org/wiki/download). Of course, I would be interested in using a more recent one!

sagotch commented:
I'll try to set up an automatic publish of latest geneweb version via appveyor in upcoming weeks.

Patriccollu commented:
@sagotch Hi Julien, have you had a chance to create a new Windows version? Thanks, Patriccollu.

sagotch commented:
Use it with caution : https://ci.appveyor.com/api/buildjobs/jwacayg7jk4poi35/artifacts/geneweb.latest.zip

I need to work a little bit more on this in order to make sure that artifacts will only come from master branch and will not be polluted by experimental builds.

NB: I do not have windows, so I did not even tried to test this release.

sagotch commented:
Note: you can check the `commit.txt` file to be sure about what you are using:

```
Commit: 1e5187d4, Fri, Nov  8, 2019  8:47:54 AM
The OCaml toplevel, version 4.09.0
Camlp5 version 7.10 (ocaml 4.09.0)
-----
```

hgouraud commented:
Note also that the bottom of the welcome page provides similar information :
`GeneWeb : compilé le 2019-11-13 à partir du commit 66f26a2c5 (2019-11-07)`

sagotch commented:
Hello @Patriccollu 

What is the status of this PR, should it be merged or not? Do you still have displaying issues with it?

Patriccollu commented:
@sagotch 
Hi Julien,

Thanks for the reminder.

I noticed that there has been a lot of changes on `lex_utf8.txt` since last November and I updated this file locally on my computer.

I need to check from scratch using the latest Windows exe - using commit https://github.com/geneweb/geneweb/commit/56c9a7f9 - and I will keep you informed.

Thanks,
Patriccollu.

Patriccollu committed:
Update lex_utf8.txt to modify Corsican translation

- Translating new strings and modifying several apostrophe signs and some quotation marks (for Corsican)
- Clarifying the use of square brackets and vertical bar (for Corsican)
- Propagating other changes done in other commits (for other languages)

Patriccollu commented:
@sagotch 
Hello Julien,

As agreed, I completed a new set of tests, especially to clarify the use of square brackets and vertical bar. Thanks to @hgouraud Henri's previous comments, I set up the following:

```
    %1 of %2
co: %1 d[i |’]%2

    (date)
co: 1<sup>u</sup> d[i']%m %y/%d d[i']%m %y/d[i']%m %y/in u %y
```

The following problem still exists but it looks to be related to template '`templm`' which is not a great problem for me:

1- With command: `http://localhost:2317/basename?lang=co;templ=templm;m=A;p=anatole;n=casata`

=> The title is correct: `Antenati d’Anatole CASATA`

2- With command: `http://localhost:2317/basename?lang=co&templ=templm&m=A&p=anatole;n=casata&maxv=5&v=4&bd=0&color=&t=G&siblings=on&alias=on&spouse=on&parents=on&rel=on&notes=on&src=on`

=> The title is incorrect: `Antenati di Anatole CASATA nnnn-nnnn`

In summary, when using the latest Windows exe [commit https://github.com/geneweb/geneweb/commit/56c9a7f9] all the other displays are correct! So this PR can be merged to take into account the changes on lex_utf8.txt.

Cheers,
Patriccollu.

hgouraud commented:
Hello Patriccollu,
there is an ongoing PR (#956) to homogeneise the issue of d[e |'] in a single form.

As for your comment about incorrect titles, it is probably due to an improper use of `%a_of_b;`
I will have a look.

hgouraud commented:
in hd/etc/templm/ancsosa.txt, line 73 and +, replace a_of_b by a_of_b2 which uses a third argument to test for vowels! Testing on the second argument is not sufficient as it may contain html decorations!!
Apparently, we missed one modification in templm!!

 ```
 %apply;a_of_b2%with;
    %if;(evar.t = "M")[*missing ancestors]%else;[*ancestors]%end;
  %and;
    %if;(not cancel_links)<a href="%prefix;%access;">%end;
      %if;(public_name != "")%public_name;%else;%first_name;%end;
      %if;(qualifier != "") <em>%qualifier;</em>%end;
      %sp;%surname;
      %if;(alias != "") <em>(%alias;)</em>%end;
    %if;(not cancel_links)</a>%end;
  %and;
    %if;(public_name != "")%public_name;%else;%first_name_key;%end;
  %end;

```

Patriccollu commented:
@hgouraud 
Hi Henri,

Great! That fixed the problem with templm.

Now the two commands have their expected titles:
```
	Antenati d’Anatole CASATA
	Antenati d’Anatole CASATA nnnn-nnnn
```

When testing, I have noticed two other problems which do not have any relation with the present one. Please can you confirm that I need to open a new issue for each one with their description?

Cheers,
Patriccollu.

hgouraud commented:
One problem per issue is preferable.

Patriccollu commented:
@sagotch 
Hi Julien,

I noticed that the check of my latest change committed 2 days ago failed!

I'm not a Github expert and I would like to have your opinion on what I have to do. Do I need to propose a new change from a brand new branch of my local repository and create a new pull request?

Thanks,
Patriccollu.

Patriccollu committed:
Merge branch 'master' into patch-2

Patriccollu commented:
@sagotch 
Hi Julien,

Actually, I just __found__ and used the `Resolve conflicts` button. I solved the 3 __simple__ conflicts indicated and clicked on `Mark as resolved`, then on `Commit merge`.

Please can you confirm that it was the right way to proceed?

Thanks,
Patriccollu.

sagotch commented:
Hi.

You should `git pull origin master --rebase` from your local repository (on branch patch-2). Be sure that your local repository is up to date with what you want to to push, and then `push -f` this branch so this PR will be updated.

You should not have a merge commit on your PR (this is automatically added by github when using the web interface :( )

Patriccollu commented:
@sagotch 
The problem is probably caused by my local repository which is not uptodate and a lot of commits is missing.

I only use the web interface. Do you know the equivalent actions - _with the web interface_ - to the commands you advised?

Thanks,
Patriccollu.

Patriccollu committed:
Update lex_utf8.txt

Patriccollu committed:
Trying to backout all my recent changes since April 16

Patriccollu committed:
Update lex_utf8.txt to modify Corsican translation

- Translating new strings and modifying several apostrophe signs and some quotation marks (for Corsican)
- Clarifying the use of square brackets and vertical bar (for Corsican)
